### PR TITLE
Sprint-8 Release

### PR DESCRIPTION
**Pull request by lpendem**
This release PR includes the following stories completed in sprint-8:

* https://github.com/nciinc/fs-permit-platform/issues/19
* https://github.com/nciinc/fs-permit-platform/issues/102
* https://github.com/nciinc/fs-permit-platform/issues/144
* https://github.com/nciinc/fs-permit-platform/issues/199
* https://github.com/nciinc/fs-permit-platform/issues/277
* https://github.com/nciinc/fs-permit-platform/issues/312
* https://github.com/nciinc/fs-permit-platform/issues/320
* https://github.com/nciinc/fs-permit-platform/issues/328
* https://github.com/nciinc/fs-permit-platform/issues/344
* https://github.com/nciinc/fs-permit-platform/issues/346
* https://github.com/nciinc/fs-permit-platform/issues/349
* https://github.com/nciinc/fs-permit-platform/issues/357

### CODE REVIEW

File: server/.eslintrc.json [hunk 1]
```diff
@@ -12,6 +12,8 @@
     "indent": ["error", 2],
     "linebreak-style": ["error", "unix"],
     "quotes": ["error", "single"],
-    "semi": ["error", "always"]
+    "semi": ["error", "always"],
+    "no-console": ["off"],
+    "no-unsafe-finally": ["off"]
```
Comment (lauraGgit) [inline server/.eslintrc.json]: Also curious about this change?

File: server/test/svg-creation.spec.es6 [hunk 2]
```diff
@@ -6,27 +6,90 @@ const expect = chai.expect;
...
-const permit = {'permitId':'8f843ab9-6a97-42f9-9e6f-c08c7fb3e6db','forestId':3,'orgStructureCode':'11-06-06','firstName':'1','lastName':'4','emailAddress':'a@a.c','treeCost':'5.00','quantity':1,'totalCost':'5.00','status':'Completed','createdAt':new Date('2018-01-19T17:15:51.435Z'),'updatedAt':new Date('2018-01-19T17:15:59.077Z'),'paygovToken':'de29d79e-aeb0-49ec-9276-b8b5c6d820e2','paygovTrackingId':'C81BBB6BED','permitExpireDate':new Date('2018-12-24T12:00:00.000Z'),'paygovError':null,'christmasTreesForest':{'id':3,'forestName':'Mt. Hood National Forest','description':'Mt. Hood | Oregon | Portland, OR','forestAbbr':'mthood','forestUrl':'https://www.fs.usda.gov/main/mthood','treeHeight':12,'stumpHeight':6,'stumpDiameter':6,'startDate':new Date('2018-11-01T12:00:00.000Z'),'endDate':new Date('2018-12-24T12:00:00.000Z'),'orgStructureCode':'11-06-06','treeCost':'5.00','maxNumTrees':5,'allowAdditionalHeight':null,'createdAt':new Date('2018-01-19T17:13:03.471Z'),'updatedAt':new Date('2018-01-19T17:13:03.471Z'),'forestNameShort':'Mt. Hood'}};
+const christmasTreesPermitFactory = require('./data/christmas-trees-permit-factory.es6');
```
Comment (lauraGgit) [inline server/test/svg-creation.spec.es6]: yay for factories!

File: wiki/christmas-trees/design_flow_process.md [hunk 3]
```diff
@@ -5,10 +5,12 @@
 2. Attach one or more screenshots of designs (as needed) to the story and uses @symbol address to notify for review.
 3. If necessary a small adhoc meeting to demo the mockups should be held.
 4. If needed, usability testing is done on the selected mockup.
-5. TTS reviews mockups and checks the "Pre approved" task which implies that the story can be deployed to staging.
+5. Throughout the sprint, TTS reviews mockups and checks the "Pre approved" task which implies that the story can be deployed to staging.
```
Comment (lauraGgit) [inline wiki/christmas-trees/design_flow_process.md]: @MelissaBraxton mind making sure this reflects our process?

Comment (MelissaBraxton) [reply wiki/christmas-trees/design_flow_process.md]: Yep, LGTM!

File: frontend/src/app/application-forms/tree-application-form/tree-application-form.component.spec.ts [hunk 4]
```diff
@@ -27,102 +30,155 @@ describe('TreeApplicationFormComponent', () => {
...
+            UtilService,
+            { provide: FormBuilder, useClass: FormBuilder },
+            { provide: Title, useClass: Title },
+            { provide: TreesService, useClass: TreesService },
+            { provide: ChristmasTreesApplicationService, useClass: MockApplicationService },
+            { provide: ApplicationFieldsService, useClass: ApplicationFieldsService },
+            {
+              provide: ActivatedRoute,
+              useValue: {
+                data: Observable.of({
```
[the remark this answers is not in the document]
Comment (lpendem) [reply frontend/src/app/application-forms/tree-application-form/tree-application-form.component.spec.ts]: @lauraGgit task added in #367

File: frontend/src/app/application-forms/tree-application-form/tree-application-form.component.spec.ts [hunk 5]
```diff
@@ -27,102 +30,155 @@ describe('TreeApplicationFormComponent', () => {
...
+      component.permit = {
+        permitId: '123',
+        firstName: 'test',
+        lastName: 'test',
+        emailAddress: 'test@test.com',
+        quantity: '2'
+      };
+      component.createForm(
+        {
+          forest: {
```
[the remark this answers is not in the document]
Comment (lpendem) [reply frontend/src/app/application-forms/tree-application-form/tree-application-form.component.spec.ts]: task in #367

File: frontend/src/app/sidebar/sidebar.component.ts [hunk 6]
```diff
@@ -25,25 +25,37 @@ export class SidebarComponent implements OnInit {
...
+    }
+  }
 
-      if (window.innerHeight < 720 && footer.getBoundingClientRect().top < 480) {
-        const bottom = -Math.abs(footer.getBoundingClientRect().top) + 840;
-        this.top = '-10px';
-        this.position = 'fixed';
-      }
+  private sidebarHeaderIntersection(container, header) {
+    if (container.getBoundingClientRect().top < this.items.length * 4.3) {
```
[the remark this answers is not in the document]
Comment (lpendem) [reply frontend/src/app/sidebar/sidebar.component.ts]: @lauraGgit styles changes based on scroll position, css does not recognize scroll position.

File: server/.eslintrc.json [hunk 7]
```diff
@@ -12,6 +12,8 @@
     "indent": ["error", 2],
     "linebreak-style": ["error", "unix"],
     "quotes": ["error", "single"],
-    "semi": ["error", "always"]
+    "semi": ["error", "always"],
+    "no-console": ["off"],
```
[the remark this answers is not in the document]
Comment (lpendem) [reply server/.eslintrc.json]: @lauraGgit we added some console errors around pay.gov integration code, needed to turn off for linter not to error out

File: server/.eslintrc.json (hunk 1 above)
Comment (lpendem) [reply server/.eslintrc.json]: @lauraGgit we have return code in `finally` that is not allowed by the linter

File: README.md [hunk 8]
```diff
@@ -222,13 +237,17 @@ VCAP service value for jwt token is used for appending a token to permit URL to
...
+##### Pay.gov error mocking in local environment
 
 * To mock a pay.gov token error, in the Christmas trees application form, set the first name to 1 and last name to 1.
 
 * To mock a pay.gov 500 error, in the Christmas trees application form, set the first name to 1 and last name to 2.
 
-* To mock a pay.gov complete collection error, use credit card number 000000000000XXXX. The last four XXXX represent the pay.gov error code.
+* To mock a pay.gov complete collection error, use credit card number `000000000000XXXX`. The last four XXXX represent the pay.gov error code.
+
+##### Pay.gov in staging environment
```
[the remark this answers is not in the document]
Comment (lpendem) [reply README.md]: @lauraGgit we are changing this in sprint-9

**Pull request merged**